src/steamship/agents/memory/chathistory.py

```diff
@@ -1,30 +1,37 @@
 from __future__ import annotations
 
-from typing import Dict, List, Optional, Union
+import uuid
+from typing import Dict, List, Optional, Union, cast
 
-from steamship import File, MimeTypes, SteamshipError
+from steamship import File, MimeTypes, Steamship, SteamshipError, Task
 from steamship.agents.schema.message_selectors import MessageSelector
 from steamship.base.client import Client
 from steamship.data import TagKind
 from steamship.data.block import Block
+from steamship.data.plugin.index_plugin_instance import EmbeddingIndexPluginInstance, SearchResults
 from steamship.data.tags import Tag
 from steamship.data.tags.tag_constants import ChatTag, DocTag, RoleTag, TagValueKey
 
 
 class ChatHistory:
-    """A ChatHistory is a wrapper of a File ideal for ongoing interactions between a user and a virtual assistant."""
+    """A ChatHistory is a wrapper of a File ideal for ongoing interactions between a user and a virtual assistant.
+    It also includes vector-backed storage for similarity-based retrieval."""
 
     file: File
+    embedding_index: EmbeddingIndexPluginInstance
 
-    def __init__(self, file: File):
+    def __init__(self, file: File, embedding_index: EmbeddingIndexPluginInstance):
         """This init method is intended only for private use within the class. See `Chat.create()`"""
         self.file = file
+        self.embedding_index = embedding_index
 
     @staticmethod
     def _get_existing_file(client: Client, context_keys: Dict[str, str]) -> Optional[File]:
         """Find an existing File object whose memory Tag matches the passed memory keys"""
-        file_query = f'kind "{TagKind.CHAT}" and name "{ChatTag.CONTEXT_KEYS}" and ' + " and ".join(
-            [f'value("{key}") = "{value}"' for key, value in context_keys.items()]
+        file_query = (
+            f'kind "{TagKind.CHAT}" and name "{ChatTag.CONTEXT_KEYS}"'
```

Comment:
The query system really comes back to pay us dividends in so many small and awesome ways

Reply:
👍